src/convert/metadataConverter.ts

```diff
@@ -120,7 +120,7 @@ export class MetadataConverter {
       const conversionPipeline = pipeline(
         Readable.from(components),
         !targetFormatIsSource && (process.env.SF_APPLY_REPLACEMENTS_ON_CONVERT === 'true' || output.type === 'zip')
-          ? (await getReplacementMarkingStream()) ?? new PassThrough({ objectMode: true })
+          ? (await getReplacementMarkingStream(defaultDirectory)) ?? new PassThrough({ objectMode: true })
```

Comment:
when doing a deploy we need to convert md from source to mdapi format so `targetFormatIsSource === true`, then `getReplacementMarkingStream` is called to return the 2nd transformer for the conversion pipeline.

`getReplacementMarkingStream` calls another func that creates an instance of `SfProject` without passing the path to the sfdx project so it defaults to `process.cwd` which isn't guaranteed to be where the project lives.